server/src/main/java/com/linecorp/centraldogma/server/metadata/RepositoryMetadataDeserializer.java

```diff
@@ -94,7 +94,16 @@ private static RepositoryRole repositoryRole(Collection<Permission> permissions)
     }
 
     private static Map<String, RepositoryRole> convert(Map<String, Collection<Permission>> permissions) {
-        return permissions.entrySet().stream()
-                          .collect(toImmutableMap(Entry::getKey, entry -> repositoryRole(entry.getValue())));
+        final Builder<String, RepositoryRole> builder = ImmutableMap.builder();
+        for (Entry<String, Collection<Permission>> entry : permissions.entrySet()) {
+            final Collection<Permission> value = entry.getValue();
+            if (!value.isEmpty()) {
+                final RepositoryRole repositoryRole = repositoryRole(value);
```

Comment:
`repositoryRole` already returns `null` if `value` is empty. How about setting `repositoryRole` if it is not null to avoid double-checking.
```java
final RepositoryRole repositoryRole = repositoryRole(value);
if (!repositoryRole) {
    builder.put(entry.getKey(), repositoryRole);
}
```

Otherwise, add a filter condition to the existing stream operations?

```java
permissions.entrySet().stream()
           .filter(sce -> !sce.getValue().isEmpty())
           .collect(toImmutableMap(Entry::getKey, entry -> repositoryRole(entry.getValue())));
```